DTS-STN/Life-Journeys
# Added geolocation functionality

# Description

Opening this to get feedback going.

[LJ-49 Geolocation ](https://lj-decd.atlassian.net/browse/LJ-49?atlOrigin=eyJpIjoiMjg3OWViM2U5NjY1NGRjMzg5YTdkNGYyYjg3NDUwYWQiLCJwIjoiaiJ9)

Added geolocation functionality to our project via an API call to [http://ip-api.com/json](http://ip-api.com/json) ONLY if the client gives permission to use their location, which is checked by using the Permissions API (more info [here](https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API/Using_the_Permissions_API) and [here](https://developers.google.com/web/updates/2015/04/permissions-api-for-the-web). I then get the region code from the JSON data, which will be used in the future to curate content based on the client's location.

**Disclaimer to be added.**

## Issues

As stated above, I am currently using IP-API to get client information. This API is free to use with a limit of 45 calls/min with no mention of any daily or monthly limits from what I can see. This seems fine for now but this is obviously not ideal once we go live. Do we have another option going forward that we could (either paid or free?)

## Test Instructions

1. Pull in branch
2. Add `console.log(data)` to the `getGeo` function
3. Type `npm run dev` and open the console to see the returned data

## Meets Definition of Done
